server/rest/src/main/java/org/infinispan/rest/resources/XSiteCacheResource.java

```diff
@@ -0,0 +1,258 @@
+            .invocation().methods(PUT).path("/v2/caches/{cacheName}/x-site/backups/{site}/take-offline-config").handleWith(this::updateTakeOffline)
+            .invocation().methods(GET).path("/v2/caches/{cacheName}/x-site/backups/{site}").withAction("cancel-receive-state").handleWith(this::cancelReceiveState)
+            .create();
+   }
+
+   private CompletionStage<RestResponse> pushStateStatus(RestRequest request) {
+      return statusOperation(request, PUSH_STATE_STATUS);
+   }
+
+   private CompletionStage<RestResponse> backupStatus(RestRequest request) {
+      return statusOperation(request, SITES_STATUS);
+   }
+
+   private CompletionStage<RestResponse> clearPushStateStatus(RestRequest restRequest) {
+      return statusOperation(restRequest, CLEAR_PUSH_STATUS);
+   }
+
+   private CompletionStage<RestResponse> cancelReceiveState(RestRequest restRequest) {
+      return executeXSiteCacheOp(restRequest, CANCEL_RECEIVE_STATE);
+   }
+
+   private CompletionStage<RestResponse> cancelPushState(RestRequest restRequest) {
+      return executeXSiteCacheOp(restRequest, CANCEL_PUSH_STATE);
+   }
+
+   private CompletionStage<RestResponse> startStatePush(RestRequest restRequest) {
+      return executeXSiteCacheOp(restRequest, PUSH_STATE);
+   }
+
+   private CompletionStage<RestResponse> takeSiteOffline(RestRequest request) {
+      return executeXSiteCacheOp(request, TAKE_OFFLINE);
+   }
+
+   private CompletionStage<RestResponse> bringSiteOnline(RestRequest request) {
+      return executeXSiteCacheOp(request, BRING_ONLINE);
+   }
+
+   private CompletionStage<RestResponse> updateTakeOffline(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+      String site = request.variables().get("site");
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+
+      XSiteAdminOperations xsiteAdmin = cache.getAdvancedCache().getComponentRegistry().getComponent(XSiteAdminOperations.class);
+
+      if (!xsiteAdmin.checkSite(site)) {
+         return CompletableFuture.completedFuture(responseBuilder.status(NOT_FOUND.code()).build());
+      }
+      byte[] byteContent = request.contents().rawContent();
+      if (byteContent == null || byteContent.length == 0) {
+         return CompletableFuture.completedFuture(responseBuilder.status(HttpResponseStatus.BAD_REQUEST.code()).build());
+      }
+
+      TakeOfflineConfiguration current = xsiteAdmin.getOfflineStatus(site).getTakeOffline();
+      TakeOffline takeOffline;
+      try {
+         takeOffline = invocationHelper.getMapper().readValue(byteContent, TakeOffline.class);
+      } catch (IOException e) {
+         return CompletableFuture.completedFuture(responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).build());
```

Comment:
shouldn't this be a `BAD_REQUEST`? 
I guess this only happens if the input is incorrect.

Reply:
Good point, I will add a test for invalid input